File: ads/src/main/java/com/ads/module/ads/wrapper/ApInterstitialAd.java
package com.ads.module.ads.wrapper;

import com.google.android.gms.ads.interstitial.InterstitialAd;

public class ApInterstitialAd {
    private InterstitialAd interstitialAd;

    public ApInterstitialAd() {
    }

    public ApInterstitialAd(InterstitialAd interstitialAd) {
        this.interstitialAd = interstitialAd;
    }

    public void setInterstitialAd(InterstitialAd interstitialAd) {
        this.interstitialAd = interstitialAd;
    }

    public boolean isReady() {
        return interstitialAd != null;
    }

    public InterstitialAd getInterstitialAd() {
        return interstitialAd;
    }
}

File: ads/src/main/java/com/ads/module/ads/wrapper/ApInterstitialAd.kt
package com.ads.module.ads.wrapper

import com.google.android.gms.ads.interstitial.InterstitialAd

class ApInterstitialAd @JvmOverloads constructor(var interstitialAd: InterstitialAd? = null) {
    val isReady: Boolean
        get() = interstitialAd != null
}
